src/Sentry/Protocol/IJsonSerializable.cs

```diff
@@ -0,0 +1,115 @@
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Text;
+using System.Text.Json;
+using Sentry.Internal.Extensions;
+
+namespace Sentry.Protocol
+{
+    internal interface IJsonSerializable
+    {
+        /// <summary>
+        /// Writes the object as JSON.
+        /// </summary>
+        void WriteTo(Utf8JsonWriter writer);
+    }
+
+    internal static class JsonSerializableExtensions
+    {
+        public static string ToJsonString(this IJsonSerializable serializable)
+        {
+            using var stream = new MemoryStream();
+            using var writer = new Utf8JsonWriter(stream);
+
+            writer.WriteSerializableValue(serializable);
+            writer.Flush();
+
+            return Encoding.UTF8.GetString(stream.ToArray());
+        }
+
+        public static void WriteSerializableValue(
+            this Utf8JsonWriter writer,
+            IJsonSerializable value)
+        {
+            value.WriteTo(writer);
+        }
+
+        public static void WriteSerializable(
+            this Utf8JsonWriter writer,
+            string propertyName,
+            IJsonSerializable value)
+        {
+            writer.WritePropertyName(propertyName);
```

Comment:
Avoid the prop altogether?

I can't think of Sentry features that accept just the key doesn't complain that there's no value